plugins/module_utils/foreman_helper.py

```diff
@@ -90,6 +92,32 @@ def inner(self, *args, **kwargs):
     return decor
 
 
+def _check_patch_needed(fixed_version=None, plugins=None):
```

Comment:
I wonder if instead of *check if required* this decorator should add those patch functions to a list (like `self._apidoc_patches`) and then `apply_apidoc_patches` becomes:

```python
def apply_apidoc_patches(self):
    for apidoc_patch in self._apidoc_patches:
        apidoc_patch()
````

Reply:
But that probably needs more thought, as the decoration happens *before* we know the server version…

Reply:
yes, maybe that is over-engineered. But should we move the "katello-patches" to the base class anyway now?

Also we could call all members starting with "_patch_" (even if that sounds rather rubyesque).

Reply:
moving katello patches is a good idea I think, yeah.